src/app/Library/CrudPanel/Traits/Create.php

```diff
@@ -220,7 +220,7 @@ private function attachManyRelation($item, $relation, $relationDetails, $relatio
         if ($relationValues === null) {
             // the developer cleared the selection
             // we gonna clear all related values by setting up the value to the fallback id, to null or delete.
-            $removedEntries = $modelInstance->where($relationForeignKey, $item->{$relationLocalKey});
+            $removedEntries = $relation;
 
             return $this->handleManyRelationItemRemoval($modelInstance, $removedEntries, $relationDetails, $relationForeignKey);
```

Comment:
```suggestion
            return $this->handleManyRelationItemRemoval($modelInstance, $relation, $relationDetails, $relationForeignKey);
```
we can pass it here.